seed_scheduler/utils.py

```diff
@@ -7,3 +9,12 @@ def get_available_metrics():
     available_metrics.extend(settings.METRICS_SCHEDULED)
 
     return available_metrics
+
+
+def calculate_retry_delay(attempt, max_delay=300):
+    """Calculates an exponential backoff for retry attempts with a small
+    amount of jitter."""
+    delay = int(random.uniform(2, 4) ** attempt)
+    if delay > max_delay:
+        delay = int(random.uniform(max_delay - 20, max_delay + 20))
```

Comment:
I didn't want to hit the exponential part after reaching the max delay. The way this works is that for any attempt that would result in over 300 seconds of delay (normally anything over 5 attempts) we use a random delay between 280 and 320 rather keeping with the exponential curve.